tests/providers/microsoft/azure/hooks/test_asb.py

```diff
@@ -19,8 +19,12 @@
 from unittest import mock
 
 import pytest
-from azure.servicebus import ServiceBusClient, ServiceBusMessage, ServiceBusMessageBatch
-from azure.servicebus.management import ServiceBusAdministrationClient
+
+try:
+    from azure.servicebus import ServiceBusClient, ServiceBusMessage, ServiceBusMessageBatch
+    from azure.servicebus.management import ServiceBusAdministrationClient
+except ImportError:
+    pytest.skip("Azure Service Bus not available", allow_module_level=True)
```

Comment:
We have everything required for install `uamqp` (requirements for `azure-servicebus`) into ARM CI image

```console
root@e3f4bad5e095:/opt/airflow# pip install azure-servicebus
Collecting azure-servicebus
  Downloading azure_servicebus-7.8.1-py3-none-any.whl (210 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 210.4/210.4 kB 2.5 MB/s eta 0:00:00
Requirement already satisfied: isodate>=0.6.0 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (0.6.1)
Requirement already satisfied: azure-core<2.0.0,>=1.14.0 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (1.26.1)
Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (4.4.0)
Requirement already satisfied: azure-common~=1.1 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (1.1.28)
Collecting uamqp<2.0.0,>=1.5.1
  Downloading uamqp-1.6.3.tar.gz (4.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.4/4.4 MB 3.9 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: six>=1.11.0 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (1.16.0)
Requirement already satisfied: msrest<2.0.0,>=0.6.17 in /usr/local/lib/python3.7/site-packages (from azure-servicebus) (0.7.1)
Requirement already satisfied: requests>=2.18.4 in /usr/local/lib/python3.7/site-packages (from azure-core<2.0.0,>=1.14.0->azure-servicebus) (2.28.1)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/site-packages (from msrest<2.0.0,>=0.6.17->azure-servicebus) (2022.12.7)
Requirement already satisfied: requests-oauthlib>=0.5.0 in /usr/local/lib/python3.7/site-packages (from msrest<2.0.0,>=0.6.17->azure-servicebus) (1.3.1)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.14.0->azure-servicebus) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.14.0->azure-servicebus) (1.26.13)
Requirement already satisfied: charset-normalizer<3,>=2 in /usr/local/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.14.0->azure-servicebus) (2.1.1)
Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.7/site-packages (from requests-oauthlib>=0.5.0->msrest<2.0.0,>=0.6.17->azure-servicebus) (3.2.2)
Building wheels for collected packages: uamqp
  Building wheel for uamqp (pyproject.toml) ... done
  Created wheel for uamqp: filename=uamqp-1.6.3-cp37-cp37m-linux_aarch64.whl size=973019 sha256=447875aa014f66c70afb9e4947df7abee8a96d0ffe35fb915fbeff1ad082b09e
  Stored in directory: /tmp/pip-ephem-wheel-cache-p3l6jc6o/wheels/29/4e/6c/a5202f34bf3871c1401f0e549dc4e275cdd80bd9dc2977bc0e
Successfully built uamqp
Installing collected packages: uamqp, azure-servicebus
Successfully installed azure-servicebus-7.8.1 uamqp-1.6.3
```

Unfortunetly `azure-servicebus` strictly prohibited for ARM in provider requirements
https://github.com/apache/airflow/blob/8e0df8881f22dd5c4c0ea71e7a9cd35b32889f47/airflow/providers/microsoft/azure/provider.yaml#L67

Reply:
Ah. Maybe it has been fixed already. Let me try it as well. Good points :)

Reply:
Actually this is a "grey zone": `uamqp` still do not have wheel packages for [aarch64](https://pypi.org/project/uamqp/#files) - open issue in Github: https://github.com/Azure/azure-uamqp-python/issues/349, so end users need to pre-install some dev packages for install it from tar.gz distribution, 

So if we change requirements for package then someone who intend to use airflow on ARM (yeah this is experimental) faced with issues to install entire provider package.